### PR TITLE
Merge tag 'android-14.0.0_r28'

### PR DESCRIPTION
**Pull request by Goooler**
## Description

- Follow up #3946.
- Diff between r21 with r28 see https://github.com/Goooler/Launcher3/compare/android-14.0.0_r21...android-14.0.0_r28

## Type of change

:white_check_mark: General change (non-breaking change that doesn't fit the below categories like copyediting)
:x: Bug fix (non-breaking change which fixes an issue)
:x: New feature (non-breaking change which adds functionality)
:x: Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/com/android/launcher3/config/FeatureFlags.java [hunk 1]
```diff
@@ -46,219 +46,273 @@
...
+    @VisibleForTesting
+    public static Predicate<BooleanFlag> sBooleanReader = f -> f.mCurrentValue;
+    @VisibleForTesting
+    public static ToIntFunction<IntFlag> sIntReader = f -> f.mCurrentValue;
+
+    private FeatureFlags() { }
+
+    public static boolean showFlagTogglerUi(Context context) {
+        return BuildConfig.IS_DEBUG_DEVICE && Utilities.isDevelopersOptionsEnabled(context);
+    }
```
Comment (Goooler) [inline src/com/android/launcher3/config/FeatureFlags.java]: Need to cherry-pick changes to FeatureFlags.

**Comment by Goooler** [comment]
CI failed due to some classes could not be found, but they have been generated, not sure why.

https://github.com/LawnchairLauncher/lawnchair/actions/runs/7831743526/job/21368797478

**Review by MrSluffy — APPROVED** [review]

**Comment by Goooler** [comment]
Reverted some format changes to avoid more merge conflicts in the future.

**Comment by MrSluffy** [comment]
All good now

**Pull request merged**